### PR TITLE
Migrate corpus pruning to engine interface.

### PR DESCRIPTION
**Pull request by oliverchang**
Rather than creating and running the LibFuzzerRunners directly.

This is the first step to making this more engine-agnostic

Also clean up various bits of minijail setup code.

### CODE REVIEW

**Comment by oliverchang** [comment]
PTAL! Will fix the CI failure tomorrow.

File: src/python/bot/tasks/corpus_pruning_task.py [hunk 1]
```diff
@@ -420,19 +406,18 @@ def process_bad_units(self, bad_units_path, quarantine_corpus_path, crashes):
...
+      try:
+        result = self._run_single_unit(unit_path)
+      except engine.TimeoutError:
         # Slow unit. Quarantine it.
         unit_path = self._quarantine_unit(unit_path, quarantine_corpus_path)
         num_bad_units += 1
         continue
 
+      if not crash_analyzer.is_memory_tool_crash(result.output):
+        # Didn't crash or time out.
```
Comment (inferno-chromium) [inline src/python/bot/tasks/corpus_pruning_task.py]: Remove "time out"

File: src/python/bot/fuzzers/engine.py [hunk 2]
```diff
@@ -126,6 +134,10 @@ def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
 
     Returns:
       A FuzzResult object.
+
+    Raises:
+      TimeoutError: If the minimization exceeds max_time.
```
Comment (inferno-chromium) [inline src/python/bot/fuzzers/engine.py]: nit: can add corpus word in these two, and for minimize_testcase, can say testcase minimization.

File: src/python/bot/tasks/corpus_pruning_task.py [hunk 3]
```diff
@@ -471,25 +456,23 @@ def run(self, initial_corpus_path, minimized_corpus_path, bad_units_path):
...
-    # corpus after merging.
-    if (result.return_code or
-        not shell.get_directory_file_count(minimized_corpus_path)):
+          'Corpus pruning timed out while merging corpus: ' + e.message)
+    except engine.MergeError as e:
+      raise CorpusPruningException('Corpus pruning failed to merge corpus: ' +
+                                   e.message)
+
+    # Sanity check that there are files in minimized corpus after merging.
+    symbolized_output = stack_symbolizer.symbolize_stacktrace(result.logs)
```
Comment (Dor1s) [inline src/python/bot/tasks/corpus_pruning_task.py]: nit: this line doesn't relate to the comment line above, maybe move it one line higher

Comment (oliverchang) [reply src/python/bot/tasks/corpus_pruning_task.py]: done.

File: src/python/bot/fuzzers/engine.py (hunk 2 above)
Comment (oliverchang) [reply src/python/bot/fuzzers/engine.py]: done!

File: src/python/bot/tasks/corpus_pruning_task.py (hunk 1 above)
Comment (oliverchang) [reply src/python/bot/tasks/corpus_pruning_task.py]: done!

**Pull request merged**